ipn/store.go

```diff
@@ -14,14 +14,79 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 	"sync"
 	"time"
 
 	"tailscale.com/atomicfile"
 	"tailscale.com/kube"
+	_ "tailscale.com/kube"
 	"tailscale.com/paths"
 )
 
+// NewStoreFunc returns a StateStore for the provided path.
+type NewStoreFunc func(path string) (StateStore, error)
+
+func init() {
+	registerDefaultStores()
```

Comment:
do this via a sync.Once and not in func init, to not slow down start-up for e.g. the tailscaled be-child case.

Reply:
Done.